src/serialize/json/normalize.ts

```diff
@@ -514,7 +517,7 @@ export class ExternalReferenceNormalizer extends BaseJsonNormalizer<Models.Exter
   normalize (data: Models.ExternalReference, options: NormalizerOptions): Normalized.ExternalReference | undefined {
     return this._factory.spec.supportsExternalReferenceType(data.type)
```

Comment:
the encoding should have happened before the `JsonSchema.isIriReference()` is run on it?
if it returns false, then `normalize` should return `undefined`

Reply:
done via https://github.com/CycloneDX/cyclonedx-javascript-library/pull/996/commits/95fb7c92e28f9bf5ed8f2378ad550f41d2f4d0bf